src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -2153,15 +2102,15 @@
   <!-- We name this "DeclarationForEachStatementSyntax" because it can express existing foreach
        loops. We may elect to represent all foreach loops using this node and deprecate (stop parsing
        into) the old one. -->
-  <Node Name="ForEachComponentStatementSyntax" Base="CommonForEachStatementSyntax">
-    <Kind Name="ForEachComponentStatement"/>
+  <Node Name="ForEachVariableStatementSyntax" Base="CommonForEachStatementSyntax">
+    <Kind Name="ForEachVariableStatement"/>
     <Field Name="ForEachKeyword" Type="SyntaxToken" Override="true">
       <Kind Name="ForEachKeyword"/>
     </Field>
     <Field Name="OpenParenToken" Type="SyntaxToken" Override="true">
       <Kind Name="OpenParenToken"/>
     </Field>
-    <Field Name="VariableComponent" Type="VariableComponentSyntax"/>
+    <Field Name="Variable" Type="ExpressionSyntax"/>
```

Comment:
><Field Name="Variable" Type="ExpressionSyntax"/> [](start = 4, length = 48)

Perhaps we can restrict the accepted expressions by listing supported kinds ```<Kind Name="...">``` inside the Field elements.